### ISSUE TITLE
Headland bug/error ?

### ISSUE THREAD

**Issue by bojanh66** [Courseplay/courseplay#6553]
Hey Tensuko, since last few releases, i have weird errors with workers and headlands..once they finish headlands, they will not start 1st row up/down, but they will go and do whole circle around the field and only then start 1st row up/down.
Also i notice course generator is creating weird headlands, before it was looking a bit different.
I did debug channels 6,12 and 14, i hope it can help.
https://youtu.be/hoW6GTf7BZ8 

![Untitled](https://user-images.githubusercontent.com/32069662/103143543-743eec00-4718-11eb-98cd-68a1609e85cf.jpg)
[log.txt](https://github.com/Courseplay/courseplay/files/5742530/log.txt)

**Comment by mricher-git** [comment]
I was going to post about the course generator as well. It would seem it always likes to start on a corner as well, in which case I need to manually turn on the combine and work into the fruit to get to the first waypoint, and then as it gets to the end of the headland, it runs down to the end and turns around to start the normal course, instead of just gracefully going into it. Please excuse the terrible drawing. I'm sure there is a good reason for it, but it feels awkward.

Starts on corner:
![course-generate2](https://user-images.githubusercontent.com/65866023/103148328-82811c80-4799-11eb-880d-5836310ceec1.jpg)

Runs to the end of the last headland and then makes a hair-pin turn moving into the normal course:
![course-generate](https://user-images.githubusercontent.com/65866023/103148337-8f9e0b80-4799-11eb-932e-c7d749cc60be.jpg)

A terrible drawing showing on the left what it is doing, and the right showing what I would expect it to do. Red, first pass, blue second headland, green, normal (up/down) course for example. The dashed line is when it makes its run to the end to start the normal course.

![course-generate3](https://user-images.githubusercontent.com/65866023/103148350-ae040700-4799-11eb-9f38-595155502770.jpg)

**Comment by Tensuko** [comment]
Do not start the course to close to a corner when use headland.

**Comment by VergylTantor** [comment]
I realize that this issue is closed but after spending a couple of hours preparing an issue report I thought that I would at least post a couple of screen shots showing the versions of generated paths that do not have this issue (6.02.00067) and the ones that do (6.02.00068+) in case some would like to generate some paths without the issue
![Screenshot (6 02 00067)](https://user-images.githubusercontent.com/39446775/103153686-ee429580-4757-11eb-9b24-3cf1f5f92190.png)
![Screenshot (6 02 00068)](https://user-images.githubusercontent.com/39446775/103153689-f1d61c80-4757-11eb-89b2-24463c149fb1.png)

**Comment by Tensuko** [comment]
Settings (and position as you see on the map)
![fsScreen_2020_12_27_13_03_19](https://user-images.githubusercontent.com/27344744/103170376-0bed2900-4844-11eb-943b-cb042a801e22.png)
even the headland change when a corner is in between got improoved
![fsScreen_2020_12_27_13_03_32](https://user-images.githubusercontent.com/27344744/103170392-21fae980-4844-11eb-963f-1444ff188e00.png)
point of where cp change to up/down rows
![fsScreen_2020_12_27_13_03_48](https://user-images.githubusercontent.com/27344744/103170427-6f775680-4844-11eb-8204-243830a43ec2.png)
this is the whole transition line and this was always the case since forever, this is nothing new.
![transition](https://user-images.githubusercontent.com/27344744/103170444-89b13480-4844-11eb-8b1e-267c3a5057ea.png)

the closer course at the change from headland1 to headland2 and so on got improoved to have better overlaps.

**Comment by pvaiko** [comment]
@mricher-git we made a change to move the transitions away from the corner. This results in what your drawing shows, please note that the one on the right side is not correct: the lower part of your dashed green line should be blue as that part of the second headland was not worked on since we transitioned to it well after the corner. So we have to move up to the transition point to finish the second headland and only then we can switch to the up/down rows, which will happen in the northwest corner.

This is intended behavior and can be tweaked by moving the starting point.

@bojanh66 we should not be making a full additional circle though, if that is the case, reopen this issue, I need the field definitions (if not on a standard map) and the settings you were using to generate the field.

Your video does not seem to correspond your drawings though, and in both cases moving the start point a bit further back should resolve your problem.

**Comment by VergylTantor** [comment]
I hope I'm not crashing this party at this point but I hope I can help with this issue. This information was taken from the Eiersholt map downloaded from Giants ModHub on field 21. https://www.farming-simulator.com/mod.php?lang=en&country=us&mod_id=180514&title=fs2019
I have included a video clip showing the machine finishing the 2nd headland pass and starting it's full circle before starting the up and down passes https://www.youtube.com/watch?v=BwwLna7nzPU&feature=youtu.be
I also added the course used and two saved games. One course is prior to starting and field work and one from close to the end of the second headland pass.
![Screenshot (17)](https://user-images.githubusercontent.com/39446775/103182462-035f1780-4871-11eb-969d-063e1cde8888.png)
[Eiersholt F21 Start.zip](https://github.com/Courseplay/courseplay/files/5745500/Eiersholt.F21.Start.zip)
[log.txt](https://github.com/Courseplay/courseplay/files/5745501/log.txt)
[CoursePlay_Courses.zip](https://github.com/Courseplay/courseplay/files/5745502/CoursePlay_Courses.zip)
[Eiersholt F21 Transition.zip](https://github.com/Courseplay/courseplay/files/5745503/Eiersholt.F21.Transition.zip)

**Comment by bojanh66** [comment]
Here is my video, take a look.
https://youtu.be/d_F_uSYs-_c 

Once the worker reach end of row, instead of making nice turn (he has plenty of free space on the headlands), he is going full circle around the field.
The only way to fix this, was to move Turn Start waypoint a bit further, maybe 3-4 meters and then he can make a nice turn.
Why does this happen ?
This happens often on pretty much every course and i have to babysit and watch workers what are they doing and edit almost every course.

**Comment by Tensuko** [comment]
@bojanh66 this is a complete other problem and got nothing to do with the coursgenerator.
Thats a Pathfinder turn.

**Comment by bojanh66** [comment]
Yes, i understand i opened this issue with name Headland bug error, because i had that error too which can be fixed by not starting a course generator at the corner...what do you say about this new issue i have with turned on "Use pathfinding in turns" ?
Is this normal or not ?

**Comment by pvaiko** [comment]
It is not, but is a different problem, open another issue.

Hate to say but I don't think anyone IRL would use a sowing machine like that (working on the headlands first and then in the middle, driving over the areas you just sowed multiple times when turning).

**Comment by bojanh66** [comment]
Yea i know, thats because headlands are selected as default when creating Course, not up/down rows and i am lazy to switch.
Tnx for explanation.

**Comment by mricher-git** [comment]
> It is not, but is a different problem, open another issue.
> 
> Hate to say but I don't think anyone IRL would use a sowing machine like that (working on the headlands first and then in the middle, driving over the areas you just sowed multiple times when turning).

True, but that would be the ideal course for a combine. I have been experimenting with moving the start location around, and it is helping, it's certainly not the obvious solution given most of the default settings allow you to choose a corner "NE/SE/etc". When I get it looking right, and it looks like it should smoothly turn into the up/down, it does the full circle. I suppose I also had use pathfinding enabled.



> please note that the one on the right side is not correct: the lower part of your dashed green line should be blue as that part of the second headland was not worked on since we transitioned to it well after the corner. So we have to move up to the transition point to finish the second headland and only then we can switch to the up/down rows, which will happen in the northwest corner.

I think you mean left, true enough, I see that now, the beginning of the dash should be solid to finish that little piece, but I still haven't wrapped my little head around why you moved away from the corner and don't go for a strategy that looks like my right-hand side diagram.
